images/router/haproxy/conf/haproxy-config.template

```diff
@@ -233,6 +233,22 @@ frontend public
   acl secure_redirect base,map_reg_int(/var/lib/haproxy/conf/os_route_http_redirect.map) -m bool
   redirect scheme https if secure_redirect
 
+    {{- range $idx, $http_request_header := .HTTPRequestHeaders }}
```

Comment:
I don't really understand why we need this on `frontend public`.  Can you help me understand?

Reply:
Also, why don't we forbid adding "Proxy" header, which we explicitly delete in https://github.com/openshift/router/pull/438/files#diff-21bd64a4ad8a7927e9d59e99e09628efc884fe6d6291e50eb33c5deb3d2a79f0R221 ?  We also make sure the "Host" is lower case, and we aren't disallowing the user to change that?

Reply:
or may be we can add this code just before https://github.com/openshift/router/pull/438/files#diff-21bd64a4ad8a7927e9d59e99e09628efc884fe6d6291e50eb33c5deb3d2a79f0R221 ?

Reply:
I have addressed this in the "Risks and Mitigations" section of the EP https://github.com/openshift/enhancements/pull/1296.